rocAL/rocAL/source/meta_data/meta_node_crop_mirror_normalize.cpp

```diff
@@ -45,48 +43,42 @@ void CropMirrorNormalizeMetaNode::update_parameters(pMetaDataBatch input_meta_da
-    vxCopyArrayRange((vx_array)_y1, 0, _batch_size, sizeof(uint),_y1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
-    vxCopyArrayRange((vx_array)_mirror, 0, _batch_size, sizeof(uint),_mirror_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_dstImgWidth, 0, _batch_size, sizeof(uint), _width_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_dstImgHeight, 0, _batch_size, sizeof(uint), _height_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_x1, 0, _batch_size, sizeof(uint), _x1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y1, 0, _batch_size, sizeof(uint), _y1_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_mirror, 0, _batch_size, sizeof(uint), _mirror_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
     for(int i = 0; i < _batch_size; i++)
     {
         auto bb_count = input_meta_data->get_labels_batch()[i].size();
         Labels labels_buf = input_meta_data->get_labels_batch()[i];
         BoundingBoxCords coords_buf = input_meta_data->get_bb_cords_batch()[i];
         BoundingBoxCords bb_coords;
-        BoundingBoxCord temp_box = {0, 0, 1, 1};
+        BoundingBoxCord temp_box = {0, 0, static_cast<float>(_width_val[i]), static_cast<float>(_height_val[i])};
         Labels bb_labels;
         BoundingBoxCord crop_box;
-        crop_box.l = (_x1_val[i]) / _src_width_val[i];
-        crop_box.t = (_y1_val[i]) / _src_height_val[i];
-        crop_box.r = (_x1_val[i] + _width_val[i]) / _src_width_val[i];
-        crop_box.b = (_y1_val[i] + _height_val[i]) / _src_height_val[i];
-        // std::cout<<"CROP Co-ordinates in CMN: lxtxrxb::\t"<<crop_box.l<<"x"<<crop_box.t<<"x"<<crop_box.r<<"x"<<crop_box.b<<"x";
+        crop_box.l = (_x1_val[i]);
+        crop_box.t = (_y1_val[i]);
+        crop_box.r = (_x1_val[i] + _width_val[i]);
+        crop_box.b = (_y1_val[i] + _height_val[i]);
         for(uint j = 0; j < bb_count; j++)
         {
-            // std::cout<<"\nIn BEFORE CMN: Box Co-ordinates lxtxrxb::\t"<<coords_buf[j].l<<"x\t"<<coords_buf[j].t<<"x\t"<<coords_buf[j].r<<"x\t"<<coords_buf[j].b<<"x\t"<<std::endl;
             if (BBoxIntersectionOverUnion(coords_buf[j], crop_box) >= _iou_threshold)
             {
                 float xA = std::max(crop_box.l, coords_buf[j].l);
                 float yA = std::max(crop_box.t, coords_buf[j].t);
                 float xB = std::min(crop_box.r, coords_buf[j].r);
                 float yB = std::min(crop_box.b, coords_buf[j].b);
-                coords_buf[j].l = (xA - crop_box.l) / (crop_box.r - crop_box.l);
-                coords_buf[j].t = (yA - crop_box.t) / (crop_box.b - crop_box.t);
-                coords_buf[j].r = (xB - crop_box.l) / (crop_box.r - crop_box.l);
-                coords_buf[j].b = (yB - crop_box.t) / (crop_box.b - crop_box.t);
+                coords_buf[j].l = (xA - crop_box.l);
```

Comment:
Check this logic once.

Reply:
Akilesh checked the outputs numerically and visually for all metanodes